elixir/apps/domain/test/domain/policies_test.exs

```diff
@@ -0,0 +1,371 @@
+
+      assert updated_policy.resource_id == policy.resource_id
+    end
+
+    test "returns error when subject has no permission to update policies", %{
+      policy: policy,
+      subject: subject
+    } do
+      subject = AuthFixtures.remove_permissions(subject)
+
+      assert update_policy(policy, %{name: "Name Change Attempt"}, subject) ==
+               {:error,
+                {:unauthorized,
+                 [
+                   missing_permissions: [
+                     {:one_of, [Policies.Authorizer.manage_policies_permission()]}
+                   ]
+                 ]}}
+    end
+
+    test "return error when subject is outside of account", %{policy: policy} do
+      other_account = AccountsFixtures.create_account()
+      other_actor = ActorsFixtures.create_actor(type: :account_admin_user, account: other_account)
+      other_identity = AuthFixtures.create_identity(account: other_account, actor: other_actor)
+      other_subject = AuthFixtures.create_subject(other_identity)
+
+      assert {:error, :unauthorized} =
+               update_policy(policy, %{name: "Should not be allowed"}, other_subject)
+    end
+  end
+
+  describe "delete_policy/2" do
+    setup context do
+      policy =
+        PoliciesFixtures.create_policy(
+          account: context.account,
+          subject: context.subject
+        )
+
+      Map.put(context, :policy, policy)
+    end
+
+    test "deletes policy", %{policy: policy, subject: subject} do
+      assert {:ok, deleted_policy} = delete_policy(policy, subject)
+      assert deleted_policy.deleted_at != nil
+    end
+
+    test "returns error when subject has no permission to delete policies", %{
+      policy: policy,
+      subject: subject
+    } do
+      subject = AuthFixtures.remove_permissions(subject)
+
+      assert delete_policy(policy, subject) ==
+               {:error,
+                {:unauthorized,
+                 [
+                   missing_permissions: [
+                     {:one_of, [Policies.Authorizer.manage_policies_permission()]}
+                   ]
+                 ]}}
+    end
+
+    test "returns error on state conflict", %{policy: policy, subject: subject} do
+      assert {:ok, deleted_policy} = delete_policy(policy, subject)
+      assert delete_policy(deleted_policy, subject) == {:error, :not_found}
+      assert delete_policy(policy, subject) == {:error, :not_found}
+    end
+
+    test "returns error when subject attempts to delete policy outside of account", %{
+      policy: policy
+    } do
+      other_account = AccountsFixtures.create_account()
+      other_actor = ActorsFixtures.create_actor(type: :account_admin_user, account: other_account)
+      other_identity = AuthFixtures.create_identity(account: other_account, actor: other_actor)
+      other_subject = AuthFixtures.create_subject(other_identity)
+
+      assert {:error, :not_found} = delete_policy(policy, other_subject)
```

Comment:
It's better to use `==` when you compare exact values and have the literal value on the right:
```suggestion
      assert delete_policy(policy, other_subject) == {:error, :not_found}
```
This is because if we use matching (`=`) then in the following cases we might have a test that won't fail if the return value changes:
```
x = %{foo: :bar}
assert %{foo: :bar} = x # true
x = Map.put(x, :fiz, "buz)
assert %{foo: :bar} = x # still true
```

I think the error message will be nicer too on failures.

Reply:
Ah, ok, this makes perfect sense.  Thank you for the examples, it definitely helps!
Still developing a sense of when matching makes sense and when it doesn't. 
Will get this updated and look to see if I did this anywhere else.

Reply:
This should be updated